test/test_log.py

```diff
@@ -80,6 +87,33 @@ def test_handler_filtering(self):
         logger.warning('bar')
         self.assertEqual(self.backend.calls(), [('WARNING', 'bar')])
 
+    def test_debug_logging(self):
+        buffer = io.StringIO()
+        with patch('sys.stderr', buffer):
+            ops.log.setup_root_logging(self.backend, debug=True)
+            logger = logging.getLogger()
+            logger.debug('debug message')
+            self.assertEqual(self.backend.calls(), [])
+            self.assertRegex(
+                buffer.getvalue(),
+                r"\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d,\d\d\d DEBUG    debug message\n")
+            logger.info('info message')
+            logger.warning('warning message')
+            logger.critical('critical message')
```

Comment:
Any patching should be done around as less code as possible (similar to try/except code blocks).

So, the subsequent `assertEqual` and `assertRegex` should be de-indented.

Reply:
The subsequent ones aren't in the with statement. And all the logging needs to happen inside the with statement.

That said, it doesn't mean a lot to assert the debug statement before doing the rest, so I moved all of the logging inside, and have only the final asserts.